### PR TITLE
chore: integrate nx into monorepo

### PR DESCRIPTION
**Pull request by Itrulia**
1. Migrated node to v16 (https://wepublish.atlassian.net/browse/WPC-680)
2. Added required files for Nx and replaced our current eslint config with Nx's. tsconfig.base.old will be replaced at a later point when all projects have been migrated
3. Did some refactorings because of the changed eslint rules
4. Migrated webpack cli to 4 as we were using the 3.x CLI with version 4

BREAKING CHANGE: Bumped `node` version to 16
BREAKING CHANGE: Bumped `graphql` version to 15
BREAKING CHANGE: Bumped `apollo-server-express` to 3
BREAKING CHANGE: `tracing` in `WepublishServerOpts` was removed

### CODE REVIEW

File: jest.preset.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const nxPreset = require('@nrwl/jest/preset').default
+
+console.log(nxPreset)
```
Comment (michael-scheurer) [inline jest.preset.js:3]: Consciously a console.log() here?

File: packages/editor/src/client/panel/subscriptionEditPanel.tsx [hunk 2]
```diff
@@ -376,233 +372,231 @@ function SubscriptionEditPanel({id, onClose, onSave}: SubscriptionEditPanelProps
...
+              {toggleRequiredLabel(t('userSubscriptionEdit.selectUser'))}
+            </Form.ControlLabel>
+
+            <UserSearch
+              name="user"
+              user={user}
+              onUpdateUser={user => {
+                setUser(user)
               }}
-              onReactivate={async () => {
```
Comment (michael-scheurer) [inline packages/editor/src/client/panel/subscriptionEditPanel.tsx:596]: Why is that being deleted?

**Review by michael-scheurer — CHANGES_REQUESTED** [review]
Amazing work. Almost no comments ;)

File: packages/editor/src/client/panel/subscriptionEditPanel.tsx (hunk 2 above)
Comment (Itrulia) [reply packages/editor/src/client/panel/subscriptionEditPanel.tsx:596]: It's not, it's just moved

File: jest.preset.js (hunk 1 above)
Comment (Itrulia) [reply jest.preset.js:3]: Yeah saw that in the meeting haha, gonna remove it with this
```suggestion

```

**Review by penk85 — DISMISSED** [review]
:+1:

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM ghcr.io/wepublish/node:latest as production
+FROM ghcr.io/wepublish/node:16.1 as production
```
[the remark this answers is not in the document]
Comment (Itrulia) [reply Dockerfile:1]: Can't wait to not use a legacy version anymore 👯

**Review by michael-scheurer — APPROVED** [review]

File: tsconfig.base.json [hunk 4]
```diff
@@ -1,32 +1,20 @@
...
+      "target": "es2015",
+      "module": "esnext",
+      "lib": ["es2017", "dom"],
+      "skipLibCheck": true,
+      "skipDefaultLibCheck": true,
+      "baseUrl": ".",
+      "paths": {}
+    },
+    "exclude": ["node_modules", "tmp"]
+  }
```
Comment (tomaszdurka) [inline tsconfig.base.json:20]: missing new line

**Review by tomaszdurka — DISMISSED** [review]
lgtm

- just one new line missing eof
- question, so we are good with 16 on website, now?

File: tsconfig.base.json (hunk 4 above)
Comment (Itrulia) [reply tsconfig.base.json:20]: Prettier says no :D

Comment (tomaszdurka) [reply tsconfig.base.json:20]: Did I lose with machine just now?

File: config/package.json [hunk 5]
```diff
@@ -2,7 +2,6 @@
   "name": "wepublish-notify",
   "private": true,
   "engines": {
-    "node": "^16.0.0",
```
Comment (Itrulia) [inline config/package.json:5]: Oops shouldn't have been in, should be reverted now

File: examples/api/src/index.ts [hunk 6]
```diff
@@ -343,7 +343,6 @@ export async function runServer() {
     urlAdapter: new ExampleURLAdapter({websiteURL}),
     playground: true,
     introspection: true,
-    tracing: true,
```
Comment (Itrulia) [inline examples/api/src/index.ts:346]: Removed as now not in Apollo anymore, could be brought back with graphql-tracing but I don't think we should have that for now

File: packages/api/__tests__/specs/article.ts [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import {ApolloServerTestClient} from 'apollo-server-testing'
+import {ApolloServer} from 'apollo-server-express'
```
Comment (Itrulia) [inline packages/api/__tests__/specs/article.ts:1]: apollo-server-testing is gone, now done this way.

File: packages/api/jest.config.js [hunk 8]
```diff
@@ -5,7 +5,7 @@ module.exports = {
   },
   globals: {
     'ts-jest': {
-      tsConfig: 'src/tsconfig.json'
+      tsconfig: 'src/tsconfig.json'
```
Comment (Itrulia) [inline packages/api/jest.config.js:8]: tsConfig was deprectaed by ts-jest

File: packages/api/package.json [hunk 9]
```diff
@@ -47,16 +47,17 @@
...
+    "graphql-upload": "^13.0.0",
     "bcrypt": "^5.0.0",
     "body-parser": "^1.19.0",
     "dataloader": "^2.0.0",
     "date-fns": "^2.28.0",
     "email-templates": "^8.0.7",
     "express": "^4.17.1",
     "form-data": "^3.0.0",
-    "graphql": "^14.6.0",
-    "graphql-iso-date": "^3.6.1",
```
Comment (Itrulia) [inline packages/api/package.json:59]: Was moved to graphql-scalars, old package does not support graphql 15

File: packages/api/src/graphql/article.ts [hunk 10]
```diff
@@ -247,107 +247,107 @@ export const GraphQLPeerArticleConnection = new GraphQLObjectType({
...
-
-          return getPublicCommentsForItemById(
-            id,
-            userSession?.user?.id ?? null,
-            null,
-            -1,
-            commentRatingSystemAnswer,
-            comment
-          )
+export const GraphQLPublicArticle: GraphQLObjectType<PublicArticle, Context> =
```
Comment (Itrulia) [inline packages/api/src/graphql/article.ts:250]: auto formatted by eslint/prettier

**Review by michael-scheurer — COMMENTED** [review]
A huge amount of work you've done. Amazing! Some comments in the code.

When starting up the application with `yarn dev` the following error occurs:
image

File: packages/editor/src/client/panel/articleMetadataPanel.tsx [hunk 11]
```diff
@@ -111,7 +111,7 @@ function ArticleMetadataPanel({value, infoData, onClose, onChange}: ArticleMetad
 
   const {t} = useTranslation()
 
-  const isAuthorized = authorise('CAN_CREATE_ARTICLE')
+  const isAuthorized = useAuthorisation('CAN_CREATE_ARTICLE')
```
[the remark this answers is not in the document]
Comment (Itrulia) [reply packages/editor/src/client/panel/articleMetadataPanel.tsx:114]: Linter complained because it's a hook it should start with `use`.

**Comment by Itrulia** [comment]
@michael-scheurer that error is expected and happens currently too. It's the API GraphQL typings that fail because the API hasn't started yet.

**Review by michael-scheurer — APPROVED** [review]

**Review by tomaszdurka — APPROVED** [review]
Please merge asap

**Pull request merged**